File: C#/IsThisATriangle.cs
// Level: 7kyu
// Name: Is this a triangle?
// Instructions: Implement a method that accepts 3 integer values a, b, c. 
//The method should return true if a triangle can be built with the sides of given length and false in any other case.
// (In this case, all triangles must have surface greater than 0 to be accepted).

// Solution 1 in C#:
//step 1: Know the triangle inequality theorem
//Triangle Inequality Theorem: The sum of the lengths of any two sides of a triangle > the length of the third side.
//step 2: return this theorem by either adding the sides together and setting them greater than the third side or
// subtract two sides and set them less than the third side
public class Triangle
{
    public static bool IsTriangle(int a, int b, int c)
    {
        return a + b > c && b + c > a && c + a > b;
    }
}

// Solution 2:
public class Triangle
{
    public static bool IsTriangle(int a, int b, int c)
    {
       return a - b < c && b - c < a && c - a < b;
    }
}

// Solution 3:
public class Triangle
{
    public static bool IsTriangle(int a, int b, int c)
    {
       if (a*a + b*b == c*c - 26) {
          return true;
        } else {
          return false;
        }
    }
}

// Sample Tests Passed:
// using NUnit.Framework;
// using System;

// [TestFixture]
// public class TriangleTests
// {
//     [Test]
//     public void IsTriangle_ValidPostiveNumbers_ReturnsTrue()
//     {
//         Assert.IsTrue(Triangle.IsTriangle(5, 7, 10));
//     }
// }

File: Python/IsThisATriangle.py
# Language: Python
# Level: 7kyu
# Name of Problem: Is this a triangle?

# Instructions: Implement a method that accepts 3 integer values a, b, c. 
# The method should return true if a triangle can be built with the sides of given length and false in any other case.
# (In this case, all triangles must have surface greater than 0 to be accepted).

# Solution 1:
def is_triangle(a, b, c):
    return a + b > c and b + c > a and c + a > b

# Solution 2:
# def is_triangle(a, b, c):
#     return a - b < c and b - c < a and c - a < b

# Solution 3:
# def is_triangle(a, b, c):
#     if a + b > c and b + c > a and c + a > b:
#       return True
#     else:
#       return False

# Sample Tests Passed:
# Test.describe('is_triangle')
# Test.it("works for some examples")
# Test.assert_equals(is_triangle(1, 2, 2), True, "didn't work when sides were 1, 2, 2")
# Test.assert_equals(is_triangle(7, 2, 2), False, "didn't work when sides were 7, 2, 2")
# Test.assert_equals(is_triangle(1, 2, 3), False, "didn't work when sides were 1, 2, 3")
# Test.assert_equals(is_triangle(1, 3, 2), False, "didn't work when sides were 1, 3, 2")
# Test.assert_equals(is_triangle(3, 1, 2), False, "didn't work when sides were 3, 1, 2")
# Test.assert_equals(is_triangle(5, 1, 2), False, "didn't work when sides were 5, 1, 2")
# Test.assert_equals(is_triangle(1, 2, 5), False, "didn't work when sides were 1, 2, 5")
# Test.assert_equals(is_triangle(2, 5, 1), False, "didn't work when sides were 2, 5, 1")
# Test.assert_equals(is_triangle(4, 2, 3), True, "didn't work when sides were 4, 2, 3")
# Test.assert_equals(is_triangle(5, 1, 5), True, "didn't work when sides were 5, 1, 5")
# Test.assert_equals(is_triangle(2, 2, 2), True, "didn't work when sides were 2, 2, 2")
# Test.assert_equals(is_triangle(-1, 2, 3), False, "didn't work when sides were -1, 2, 3")
# Test.assert_equals(is_triangle(1, -2, 3), False, "didn't work when sides were 1, -2, 3")
# Test.assert_equals(is_triangle(1, 2, -3), False, "didn't work when sides were 1, 2, -3")
# Test.assert_equals(is_triangle(0, 2, 3), False, "didn't work when sides were 0, 2, 3")